src/Components/SmartComponents/Modals/OsExposureReportModal/OsExposureReportModal.js

```diff
@@ -0,0 +1,234 @@
+import React, { useEffect, useState } from 'react';
+import propTypes from 'prop-types';
+import {
+    Modal,
+    Button,
+    Form,
+    FormGroup,
+    TextInput,
+    Flex,
+    FlexItem,
+    Checkbox,
+    TextArea
+} from '@patternfly/react-core';
+import LifecycleSelect from './LifecycleSelect';
+import isEqual from 'lodash/isEqual';
+import { CUSTOM_REPORT_FILTER_DATA_BY_LABEL, getCveReportFilters } from '../../../../Helpers/constants';
+
+const OsExposureReportModal = ({
+    isOsExposureModalOpen,
+    setOsExposureModalOpen,
+    osExposureData
+}) => {
+    const [reportTitle, setReportTitle] = useState('');
+    const [exposureMap, setExposureMap] = useState({});
+    const [selectedOsVersions, setSelectedOsVersions] = useState([]);
+    const [filterData, setFilterData] = useState({
+        impact: [],
+        advisory_available: ['true']
+    });
+    const [userNote, handleUserNoteChange] = useState('');
+    const CVE_REPORT_FILTERS = getCveReportFilters(false, true);
+    const ACTIVE_FILTERS = [
+        'impact',
+        'advisory_available'
+    ];
+
+    useEffect(() => {
+        if (osExposureData && osExposureData.length) {
+            const exposureMapCopy = { ...exposureMap };
+            osExposureData.sort((a, b) => {
+                if (a.major < b.major) {return 1;}
+
+                if (a.major > b.major) {return -1;}
+
+                if (a.minor < b.minor) {return 1;}
+
+                if (a.minor > b.minor) {return -1;}
+            }).forEach((os) => {
+                const majorVer = `${os.name} ${os.major}`;
+                const minorVer = `${os.major}.${os.minor}`;
+                const lifecycle = `${os.lifecycle_phase}`;
+
+                if (!exposureMapCopy[majorVer]) {
+                    exposureMapCopy[majorVer] = {};
+                }
+
+                if (!exposureMapCopy[majorVer][minorVer]) {
+                    exposureMapCopy[majorVer][minorVer] = {};
+                }
+
+                exposureMapCopy[majorVer][minorVer][lifecycle] = os;
+            });
+
+            setExposureMap(exposureMapCopy);
+        }
+    }, [osExposureData]);
+
+    const isSameOsMinor = (osMinor) => {
+        let isSelected = false;
+        selectedOsVersions.forEach((os) => {
+            if (isEqual(os, osMinor)) {
+                isSelected = true;
+                return;
```

Comment:
Returning `undefined` from a function which should return a boolean.

Function could be improved as such:

```js
const isMinorOsVersionSelected = (targetOsVersion) =>
    selectedOsVersions.some(osVersion => isEqual(osVersion, targetOsVersion))
```

Reply:
How is this improved?

Reply:
The code simplified, easier to read and avoids confusion when a function that seems to return a boolean returns undefined.